vault/resource_ldap_secret_backend_static_role.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/util"
+)
+
+func ldapSecretBackendStaticRoleResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
+		consts.FieldMount: {
+			Type:         schema.TypeString,
+			Default:      consts.MountTypeLDAP,
+			Optional:     true,
+			Description:  "The path where the LDAP secrets backend is mounted.",
+			ValidateFunc: provider.ValidateNoLeadingTrailingSlashes,
+		},
+		consts.FieldRole: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Name of the role.",
+			ForceNew:    true,
+		},
+		consts.FieldUsername: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "The username of the existing LDAP entry to manage password rotation for.",
+			ForceNew:    true,
+		},
+		consts.FieldDN: {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Distinguished name (DN) of the existing LDAP entry to manage password rotation for.",
+		},
+		consts.FieldRotationPeriod: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "How often Vault should rotate the password of the user entry.",
+		},
+	}
+	return provider.MustAddMountMigrationSchema(&schema.Resource{
+		CreateContext: createLDAPStaticRoleResource,
+		UpdateContext: createLDAPStaticRoleResource,
+		ReadContext:   ReadContextWrapper(readLDAPStaticRoleResource),
+		DeleteContext: deleteLDAPStaticRoleResource,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		CustomizeDiff: getMountCustomizeDiffFunc(consts.FieldMount),
+		Schema:        fields,
+	})
+}
+
+func createLDAPStaticRoleResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mount := d.Get(consts.FieldMount).(string)
+	role := d.Get(consts.FieldRole).(string)
+	rolePath := fmt.Sprintf("%s/static-role/%s", mount, role)
+	log.Printf("[DEBUG] Creating LDAP static role at %q", rolePath)
+	data := map[string]interface{}{}
+	fields := []string{
```

Comment:
nit: we could consider making this variable package-level to be reused for the create and read